ccswbs/doorknob
# First prototype of program search page

A first prototype of the program search page.

Right now the prototype uses a yaml file rather than pulling from content hub.

I scraped the data from calendar.uoguelph.ca for the yaml file so it is not perfect and isn't an indicator of what will appear on the real page. It is simply just an example of how the data might look like/be structured.

I've setup the GraphQL query similarly to how we did with the spotlight search so it should be simple to switch data sources when we want to pull from content hub.

# Summary of changes

## Frontend
- Added new pages: undergraduate-programs.js and graduate-programs.js
- Added new components: programCard.js, programSearch.js, linkTabs.js and programTabs.js
- Added new stylesheet: program-search.scss (with the intention of switching to tailwind later and removing this)
- Added new yaml files: undergraduate-programs.yml and graduate-programs.yml

### Incidental changes
- N/A

## Backend
- N/A

## Test plan
- PR build: https://deploy-preview-37--doorknob.netlify.app/

1. Go to https://deploy-preview-37--doorknob.netlify.app/undergraduate-programs
2. Enter text in the search bar and note what programs appear and whether it makes sense.
3. Use the filter by type feature and see whether programs are correctly filtered.
4. Repeat 1 - 3 on https://deploy-preview-37--doorknob.netlify.app/graduate-programs
